File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) 
    {
        string str="";
        for(int i=0;i<s.size();i++)
        {
            if(s[i]>='A' && s[i]<='Z')
            {
                str += tolower(s[i]);
            }
            else if(s[i]>='a' && s[i]<='z')
            {
                str += s[i];
            }
            else if(s[i]>='0' && s[i]<='9')
            {
                str += s[i];
            }
        }
        for(int i=0;i<str.size()/2;i++)
        {
            if(str[i]!=str[str.size()-1-i])
            {
                return false;
            }
        }
        return true;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder sb = new StringBuilder();
        int n = s.length();
        for(int i=0;i<n;i++){
            if(s.charAt(i)>='A' && s.charAt(i)<='Z'){
                sb.append(Character.toLowerCase(s.charAt(i)));
            }
            else if(s.charAt(i)>='a' && s.charAt(i)<='z'){
                sb.append(s.charAt(i));
            }
            else if(s.charAt(i)>='0' && s.charAt(i)<='9'){
                sb.append(s.charAt(i));
            }
        }
        int left=0;
        int right=sb.length()-1;
        while(left<=right){
            if(sb.charAt(left)!=sb.charAt(right)){
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
}